File: Challenges/2/insertShiftArray.py
def insertShiftArray (arr, num) :
  if len(arr) % 2 == 0 :
    middle = len(arr) / 2 
  else :
    middle = int(len(arr) / 2) + 1

  newArr = []

  i = 0

  while (i < len(arr)) :
    if i == middle and newArr[len(newArr) - 1] != num:
      newArr.append(num)
    else :
     newArr.append(arr[i])
     i+=1

  return newArr

File: Challenges/02/insertShiftArray.js
const insertShiftArray = (arr, num) => {
  const middle = arr.length % 2 === 0 ? arr.length / 2 : parseInt(arr.length / 2) + 1;
  const newArr = [];

  let i = 0;

  while (i < arr.length) {
    (i === middle && newArr[newArr.length - 1] !== num) ? newArr.push(num) : newArr.push(arr[i++]);
  };

  return newArr;
};